daal4py/sklearn/neighbors/_base.py

```diff
@@ -340,19 +340,26 @@ def _fit(self, X, y=None):
 
         weights = getattr(self, 'weights', 'uniform')
 
-        if daal_check_version(((2020,'P', 3),(2021,'B', 110))) and not X_incorrect_type \
-        and weights in ['uniform', 'distance'] and self.algorithm in ['brute', 'kd_tree', 'auto', 'ball_tree'] \
-        and (self.metric == 'minkowski' and self.p == 2 or self.metric == 'euclidean') \
-        and single_output and fptype is not None and not sp.issparse(X) and correct_n_classes:
-            logging.info("sklearn.neighbors.NeighborsBase._fit: " + method_uses_daal)
-            daal4py_fit(self, X, fptype)
-            result = self
-        else:
+        def stock_fit(self, X, y):
             logging.info("sklearn.neighbors.NeighborsBase._fit: " + method_uses_sklearn)
             if SKLEARN_24:
                 result = super(NeighborsBase, self)._fit(X, y)
             else:
                 result = super(NeighborsBase, self)._fit(X)
+            return result
+
+        if daal_check_version(((2020,'P', 3),(2021,'B', 110))) and not X_incorrect_type \
+        and weights in ['uniform', 'distance'] and self.algorithm in ['brute', 'kd_tree', 'auto', 'ball_tree'] \
+        and (self.metric == 'minkowski' and self.p == 2 or self.metric == 'euclidean') \
+        and single_output and fptype is not None and not sp.issparse(X) and correct_n_classes:
+            try:
+                daal4py_fit(self, X, fptype)
+                result = self
+                logging.info("sklearn.neighbors.NeighborsBase._fit: " + method_uses_daal)
+            except RuntimeError:
+                result = stock_fit(self, X, y)
```

Comment:
we use message for this case `method_uses_sklearn_arter_daal`. see linear_regression for example